packages/flutter_tools/lib/src/update_packages_pins.dart

```diff
@@ -23,6 +23,8 @@ const kManuallyPinnedDependencies = <String, String>{
   'archive': '3.6.1', // https://github.com/flutter/flutter/issues/115660
   'code_assets':
       '0.19.4', // Under active development with breaking changes until 1.0.0. Manually rolled by @dcharkes.
+  'dds':
+      '5.0.3', // 5.0.4 contains bugs described in https://github.com/Dart-Code/Dart-Code/issues/4678.
```

Comment:
This isn't quite correct, 5.0.4 contains the _fix_ for that issue, but it triggers a leak of scripts in the VM so this reverts that fix for now.